exporter/containerimage/opts.go

```diff
@@ -96,9 +95,36 @@ func (c *ImageCommitOpts) Load(opt map[string]string) (map[string]string, error)
 		c.OCITypes = true
 	}
 
+	c.AddAnnotations(as)
+
 	return rest, nil
 }
 
+func (c *ImageCommitOpts) AddAnnotations(annotations AnnotationsGroup) {
+	if annotations == nil {
+		return
+	}
+	if c.Annotations == nil {
+		c.Annotations = AnnotationsGroup{}
+	}
+	c.Annotations = c.Annotations.Merge(annotations)
+	for _, a := range annotations {
+		if len(a.Index)+len(a.IndexDescriptor)+len(a.ManifestDescriptor) > 0 {
+			if !c.OCITypes {
+				logrus.Warn("forcibly turning on oci-mediatype mode for annotations")
+				c.OCITypes = true
+			}
+		}
+	}
+}
+
+func (c *ImageCommitOpts) EnableAttestations() {
```

Comment:
Why is this called `EnableAttestations` and not `EnableOCITypes` (if the helper is needed at all). These opts don't know about attestations afaics.

Reply:
Have reworked the helper to `EnableOCITypes` that takes a `reason` as a log message. The helper is useful to get consistent warning messages for the different reasons we might change the oci mediatypes option, so I think it's worth keeping.